query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/mod.rs

```diff
@@ -284,52 +308,47 @@ impl fmt::Display for ConnectorVersion {
-
     if !capabilities.is_empty() && !connector.capabilities().contains(capabilities) {
         println!("Connector excluded. Missing required capability.");
         return false;
     }
 
-    // We skip tests that exclude JS driver adapters when an external test executor is configured.
-    // A test that you only want to run with rust drivers can be annotated with exclude(JS)
-    if CONFIG.external_test_executor().is_some() && exclude.iter().any(|excl| excl.0.to_uppercase() == "JS") {
-        println!("Excluded test execution for JS driver adapters. Skipping test");
-        return false;
-    };
-    // we consume the JS token to prevent it from being used in the following checks
-    let exclude: Vec<_> = exclude.iter().filter(|excl| excl.0.to_uppercase() != "JS").collect();
-
-    // We only run tests that include JS driver adapters when an external test executor is configured.
-    // A test that you only want to run with js driver adapters can be annotated with only(JS)
-    if CONFIG.external_test_executor().is_none() && only.iter().any(|incl| incl.0.to_uppercase() == "JS") {
-        println!("Excluded test execution for rust driver adapters. Skipping test");
-        return false;
-    }
-    // we consume the JS token to prevent it from being used in the following checks
-    let only: Vec<_> = only.iter().filter(|incl| incl.0.to_uppercase() != "JS").collect();
+    let exclusions = exclude
+        .iter()
+        .filter_map(|c| ConnectorVersion::try_from(*c).ok())
+        .collect::<Vec<_>>();
+
+    let inclusions = only
+        .iter()
+        .filter_map(|c| ConnectorVersion::try_from(*c).ok())
+        .collect::<Vec<_>>();
 
-    if !only.is_empty() {
-        return only
-            .iter()
-            .any(|incl| ConnectorVersion::try_from(**incl).unwrap().matches_pattern(&version));
+    for exclusion in exclusions.iter() {
+        for inclusion in inclusions.iter() {
+            if exclusion.is_broader(inclusion) {
+                panic!("Error in connector test execution rules. Version `{exclusion}` in `excluded()` is broader than `{inclusion}` in `only()`");
+            }
+        }
     }
 
-    if exclude.iter().any(|excl| {
-        ConnectorVersion::try_from(**excl)
-            .map_or(false, |connector_version| connector_version.matches_pattern(&version))
-    }) {
+    if exclusions.iter().any(|excl| excl.matches_pattern(version)) {
         println!("Connector excluded. Skipping test.");
         return false;
     }
 
+    if !inclusions.is_empty() {
+        return inclusions.iter().any(|incl| incl.matches_pattern(version));
+    }
+
     // FIXME: This skips vitess unless explicitly opted in. Replace with `true` when fixing
     // https://github.com/prisma/client-planning/issues/332
-    !matches!(version, ConnectorVersion::Vitess(_))
+    CONFIG.external_test_executor().is_some() || !matches!(version, ConnectorVersion::Vitess(_))
```

Comment:
So what happens to https://github.com/prisma/client-planning/issues/332 and `!matches!(version, ConnectorVersion::Vitess(_))` ? Is it still even called at all?

Reply:
> Part of https://github.com/prisma/team-orm/issues/541

It only focuses on driver adapters, not on all the tests. Fixing the vitess bigger issue needs more work, to be done as part of tech debt.

The new clause is an `||` i.e it's less restrictive. Before it didn't run for vitess. Now it doesn't run for vitess, except if it's driver adapters, case in which it also runs.